python/cuml/cuml/manifold/umap.pyx

```diff
@@ -879,6 +879,10 @@ class UMAP(Base,
                 "using nn_descent as build_algo on a small dataset (< 150 samples) is unstable"
             )
 
+        if self.n_rows <= 300:
+            # non-deterministic calculations do not work with limited number of samples
+            self.deterministic = True
```

Comment:
We are trying to get closer to scikit-learn in terms of API. Let's not add more attributes that don't fit with that.

Fitted attributes (those determined as part of `fit`) like this one should end in a `_` or if you don't want to expose this as a fitted attribute then it should start with a `_`. "Vanilla" attributes (`self.foo`) should only exist if there is a constructor argument that matches them, and their value should not be changed

Reply:
We could move it to `deterministic_` and call it a legitimate bug fix

Reply:
I just updated the attribute with a private prefix. Do we want to keep this or make the overall attribute name change in a separate follow-up PR?

Reply:
Please do not change the attribute name in this PR or any follow-up PR. We are going to address this issue more broadly within the scope of https://github.com/rapidsai/cuml/issues/6432 .